app/models/article.rb

```diff
@@ -35,7 +35,9 @@ def name
   end
 
   def path
-    if published?
+    if published? && short_path
```

Comment:
Ah, yes so it's not replacing but just being an automator of sorts for making redirects based on the field. Got it.